Library/Homebrew/extend/ENV/super.rb

```diff
@@ -111,16 +111,17 @@ def determine_path
 
     # Homebrew's apple-gcc42 will be outside the PATH in superenv,
     # so xcrun may not be able to find it
-    case homebrew_cc
-    when "gcc-4.2"
-      begin
+    begin
+      case homebrew_cc
+      when "gcc-4.2"
         apple_gcc42 = Formulary.factory("apple-gcc42")
-      rescue FormulaUnavailableError
+        paths << apple_gcc42.opt_bin.to_s if apple_gcc42
+      when GNU_GCC_REGEXP
+        gcc_formula = gcc_version_formula($&)
+        paths << gcc_formula.opt_bin.to_s if gcc_formula
       end
-      paths << apple_gcc42.opt_bin.to_s if apple_gcc42
-    when GNU_GCC_REGEXP
-      gcc_formula = gcc_version_formula($&)
-      paths << gcc_formula.opt_bin.to_s
+    rescue FormulaUnavailableError
+      nil
```

Comment:
I've added the comment. The comment doesn't suppress the `brew style` warning. Should I also include `nil` to suppress the `brew style` warning?